Issue#247: Fix default examples and models

Vladyslav-Frolov opened the pull request:
dev
## ZenHub

* [Main ZenHub ticket](https://app.zenhub.com/workspaces/home-project-5f7b77ff8db73a001cb17009)


## Code reviewers

- [x] @kh0ma
- [x] @likeRewca
- [ ] @Ilya-Ross
- [ ] @BenAvleck
- [ ] @AriannaLiss
- [ ] @mitianin
- [x] @Illia-Shchur
- [x] @IhorSamoshost
- [x] @kizimov

## Summary of issue

Fix default models and examples for creating/updating users, cooperations etc.

As an admin I want to have correct examples for creating and updating users, cooperation etc.

## Summary of change

In schema "updateUser" deleted "allOf reference" to id, fixed reference in apartment tag. Added examples for a list of houses. Deleted ID in contacts of email in the example of creating a user on 1481 row. Updated the user --> the ID was removed from the payload because it is in the path variable. In examples --> added quotes for the string data type

## Testing approach

Tested using a full deployment cycle via docker

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

Vladyslav-Frolov committed:
Issue#247-Fix default examples and models --> fix: delete row wih admin_email from ReadCooperation model, because it redundant // updateUser --> delete allOf -ref to id // fix ref in apartment tag // the tests are running correctly / add examples for list of houses // delete id in the example of creating a user in contacts of email 1481 row // update the user --> the ID was removed from the payload, because it is in the path variable // example --> added quotes for the integer data type

Vladyslav-Frolov committed:
Issue#247-Fix default examples and models --> fix: add description for tags in yaml file

likeRewca reviewed (changes requested):
Create contact - /users/{user_id}/contasts - example is not correct(only type and main fields) on request and answer 201. Just look all answer on contacts end point.
Apartmen - houses/{house_id}/apartment the same prople on contacts on answers.
Cooperation contacts - the same.

Vladyslav-Frolov committed:
Issue#247-Fix default examples and models --> fix: Added: add mail field to CreateContact and ReadContact

kh0ma commented on home-open-api/src/main/resources/yaml/openapi.yaml:
```diff
@@ -36,6 +36,16 @@ tags:
+
+
+
+
```
Why do we need these extra lines?

Vladyslav-Frolov replied:
I agree and have already corrected it

kh0ma commented on home-open-api/src/main/resources/yaml/openapi.yaml:
```diff
@@ -1456,21 +1466,21 @@ components:
         title:
           type: string
-          example: News In The Local Area
+          example: "News In The Local Area"
```
Honestly, I think that quotes here bring any visibility neither readability. Considering yaml specification they are redundant and I suggest removing them.

Vladyslav-Frolov replied:
I’m not so sure. In my experience, and other team members also agree that examples taken in quotation marks are marked in green (depends on the editor), they are visually easy to search for. Without quotes, they merge with the general text. I suggest leaving the quotes for the examples. We can also bring this question for discussion to the next meeting.

kh0ma commented on home-open-api/src/main/resources/yaml/openapi.yaml:
```diff
@@ -1584,25 +1593,23 @@ components:
-            - id: 1
-            - main: false
-            - email: "readUserEmail@example.com"
+              id: 1
```
Can we make examples closer to the backend app we are implementing? I mean if we are using aplication/json mediatype then examples should be of the same format.

Vladyslav-Frolov replied:
Perhaps I didn't fully understand the question. But in the end, this example generates a JSON file the same as we get in a real application.

kh0ma replied:
I agree that JSON and YAML are interchangeable here. But when we briefly look at the code it would be great to see json here without additional steps to go to the converter and make json from yaml. Please define all defined examples as json.

Vladyslav-Frolov replied:
> I agree that JSON and YAML are interchangeable here. But when we briefly look at the code it would be great to see json here without additional steps to go to the converter and make json from yaml. Please define all defined examples as json.

already done

Vladyslav-Frolov committed:
Issue#247-Fix default examples and models --> fix:
Added: field email to UpdateContact in yaml with example
Added: examples of email and phone to ReadApartment
Changed: increment version of open-api spec.

Vladyslav-Frolov committed:
Issue#247-Fix default examples and models --> fix: Changed: removed the extra lines

Vladyslav-Frolov commented:
@likeRewca 
> Create contact - /users/{user_id}/contasts - example is not correct(only type and main fields) on request and answer 201. Just look all answer on contacts end point.
> Apartmen - houses/{house_id}/apartment the same prople on contacts on answers.
> Cooperation contacts - the same.

This fix is done

IhorSamoshost commented on home-open-api/src/main/resources/yaml/openapi.yaml:
```diff
@@ -1745,7 +1763,7 @@ components:
           type: string
-          example: test.receive.messages@gmail.com
+          example: "test.receive.messages@gmail.com"
     UpdatePhoneContact:
```
1) Am I correct in understanding that when creating or updating an email contact, you need to enter email twice - one as a property of the base model and the other as a property of the child model?
2) In all the models above, we set the number of characters for e-mail from 5 to 320, and here - from 9 to 320. Is this a typo?

Vladyslav-Frolov replied:
> 1. Am I correct in understanding that when creating or updating an email contact, you need to enter email twice - one as a property of the base model and the other as a property of the child model?
> 2. In all the models above, we set the number of characters for e-mail from 5 to 320, and here - from 9 to 320. Is this a typo?

1. It looks like a mistake, give me some time to find out
2. No, it isn't a typo, that's the way it's meant to be

Vladyslav-Frolov replied:
> 1. Am I correct in understanding that when creating or updating an email contact, you need to enter email twice - one as a property of the base model and the other as a property of the child model?

After looking at it with a more fresh glance, I corrected this reappearance

IhorSamoshost commented on home-open-api/src/main/resources/yaml/openapi.yaml:
```diff
@@ -2210,9 +2251,11 @@ components:
                     main: false
+                    phone: "+380506617306"
               ownership_part: 0.38
 
```
Is it necessary that in the example the sum of the ownership shares does not exceed 1?

Vladyslav-Frolov replied:
> Is it necessary that in the example the sum of the ownership shares does not exceed 1?

this question is beyond the scope of my assignment, it is better to ask it during a meeting with a technical expert and / or business analysts

kh0ma replied:
@IhorSamoshost Yes, we have such validation.

Vladyslav-Frolov committed:
Issue#247-Fix default examples and models --> fix:
Fixed: removed the extra fields and added full examples in the openapi.yaml specification models, namely: ReadContact CreateContact UpdateContact

Vladyslav-Frolov committed:
Issue#247-Fix-default-examples-and-models --> fix:
Fixed: update minLength from 9 to 5 for email for UpdateEmailContact

likeRewca commented:
[Contact](https://prnt.sc/1asbk5o) still the same example

Vladyslav-Frolov committed:
Issue#247-Fix-default-examples-and-models --> fix:
Changed: all examples converted from YAML to JSON representation where it needs
Add: example in InvitationType

Vladyslav-Frolov committed:
Issue#247-Fix-default-examples-and-models --> fix:
Fixed: a fragment of the test message that was 9 became 5 minimum letter size in updateContactEmailWithEmptyParameterTest() method of ContactApiIT class.  This became a necessity after the changes of openapi.yaml

kh0ma commented on home-open-api/src/main/resources/yaml/openapi.yaml:
```diff
@@ -8,7 +8,7 @@ info:
   contact:
     url: https://github.com/ita-social-projects/Home
-  version: 1.0.0
+  version: 1.0.1
```
I'm not sure that we've started bumping this version. Let's postpone it until we decide on our versioning. Please revert.

Vladyslav-Frolov replied:
> I'm not sure that we've started bumping this version. Let's postpone it until we decide on our versioning. Please revert.

ok, I will revert

kh0ma commented on home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/contacts/ContactApiIT.java:
```diff
@@ -1,38 +1,24 @@
-import com.softserveinc.ita.homeproject.model.UpdateContact;
-import com.softserveinc.ita.homeproject.model.UpdateEmailContact;
-import com.softserveinc.ita.homeproject.model.UpdatePhoneContact;
+import com.softserveinc.ita.homeproject.model.*;
```
Please do not use import suppression in production code.

Vladyslav-Frolov replied:
> Please do not use import suppression in production code.

ok, I will revert

kh0ma commented on home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/contacts/ContactApiIT.java:
```diff
@@ -1,38 +1,24 @@
+import static com.softserveinc.ita.homeproject.api.tests.utils.ApiClientUtil.BAD_REQUEST;
+import static com.softserveinc.ita.homeproject.api.tests.utils.ApiClientUtil.NOT_FOUND;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.junit.jupiter.api.Assertions.*;
```
The same.

Vladyslav-Frolov replied:
> The same.

ok, I will revert

Vladyslav-Frolov committed:
Merge branch 'dev' into Issue#247-Fix-default-examples-and-models

kh0ma commented on home-open-api/src/main/resources/yaml/openapi.yaml:
```diff
@@ -36,6 +36,12 @@ tags:
     description: Cooperation poll management operations
   - name: poll
     description: Poll management operations  independently from cooperation
+  - name: cooperation contact
```
Do we really need these new tags?

Vladyslav-Frolov replied:
> Do we really need these new tags?

This is already existing tag that was previously forgotten to be added to the list of common tags. I just added it to the "tag" section and added a description that was not there before.

Vladyslav-Frolov committed:
Issue#247-Fix-default-examples-and-models --> fix:
Reverted: all implicit imports (with *) in ContactApiIT
Reverted: version in openapi.yaml

Vladyslav-Frolov committed:
Issue#247-Fix-default-examples-and-models --> fix:
Added: example to QuestionType
